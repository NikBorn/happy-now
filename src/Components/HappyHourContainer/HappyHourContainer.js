import React from 'react';
import PropTypes from 'prop-types';
import HappyHourCard from '../HappyHourCard/HappyHourCard.js';
import LoadingScreen from '../LoadingScreen/LoadingScreen';

const HappyHourContainer = (props) => {

  const happyHoursArray = props.locations.map(happyHour=> {
    return (
      <HappyHourCard key={happyHour.locationId} happyHourInfo={happyHour} />
    );
  });

  const happyHourDisplay = props.locations.length 
    ? happyHoursArray 
    : 
    <LoadingScreen />;

  return (
    <div className='happy-hour-cntr'>
      { happyHourDisplay }
    </div>
  );
};

HappyHourContainer.propTypes = {
  locations: PropTypes.array
};


export default HappyHourContainer;