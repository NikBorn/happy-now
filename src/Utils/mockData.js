

export const mockLocationsResponse = [
  {
    isFavorite: false,
    id: "4b044f71f964a5202e5322e3",
    name: "The Glenn",
    contact: {
      phone: "3032551400",
      formattedPhone: "(303) 255-1400",
      twitter: "biggyglenn",
      facebook: "115875251829275",
      facebookName: "Glenn Bar"},
    location: {
      address: "11140 Irma Dr",
      lat: 39.89855629220959,
      lng: -104.96589290871326,
      labeledLatLngs: [{
        label: "display",
        lat: 39.89855629220959,
        lng: -104.96589290871326
      }],
      distance: 4876,
      postalCode: "80233",
      cc: "US",
      city: "Northglenn",
      state: "CO",
      country: "United States",
      formattedAddress: ['11140 Irma Dr', 'Northglenn, CO 80233', 'United States']},
    categories: [{
      id: "4bf58dd8d48988d116941735",
      name: "Bar",
      pluralName: "Bars",
      shortName: "Bar",
      icon: {
        prefix: "https://ss3.4sqi.net/img/categories_v2/nightlife/pub_",
        suffix: ".png"
      },
      primary: true}],
    verified: false,
    stats: {
      checkinsCount: 2080,
      usersCount: 702,
      tipCount: 29},
    url: "http://theglennbarandgrill.com",
    price: {
      tier: 2,
      message: "Moderate",
      currency: "$"},
    hasMenu: true,
    rating: 8.4,
    ratingSignals: 84,
    menu: { 
      type: "Menu", 
      label: "Menu", 
      anchor: "View Menu",
      url: "https://foursquare.com/v/the-glenn/4b044f71f964a5202e5322e3/menu",
      mobileUrl: "https://foursquare.com/v/4b044f71f964a5202e5322e3/device_menu"},
    allowMenuUrlEdit: true,
    beenHere: {
      count: 0,
      marked: false,
      lastCheckinExpiredAt: 0},
    hours: {
      status: "Likely open",
      richStatus: {
        entities: [],
        text: "Likely open"},
      isOpen: true,
      isLocalHoliday: false},
    photos: {
      count: 0,
      groups: []},
    hereNow: {
      count: 0,
      summary: "Nobody here",
      groups: [] }},
  {
    name: "Tequila Rock",
    price: { tier: 2, message: "Moderate", currency: "$" },
    rating: 6.4,
    ratingSignals: 399,
    contact: { phone: "3032551400", 
      formattedPhone: "(303) 255-1400", 
      twitter: "biggyglenn", 
      facebook: "115875251829275", 
      facebookName: "Glenn Bar" },
    isFavorite: false,
    id: "4d8a35c976e1236a99721534"
  },
  {
    name: "Paddy's Pub",
    price: { tier: 1, message: "Moderate", currency: "$" },
    rating: 3.4,
    ratingSignals: 107,
    contact: { phone: "3038675309", 
      formattedPhone: "(303) 867-5309", 
      twitter: "biggyglenn", 
      facebook: "115875251829275", 
      facebookName: "Glenn Bar" },
    isFavorite: false,
    id: "4d8a35c976e1236a99725553"
  }
];

