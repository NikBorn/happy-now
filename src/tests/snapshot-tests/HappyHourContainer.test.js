import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import HappyHourContainer from 
  '../../Components/HappyHourContainer/HappyHourContainer.js';
import { mockHappyHourInfo } from '../../Utils/mockData.js';
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() });

describe('HappyHourContainer', () => {
  it('should match the snapshot everytime', () => {
 
    const wrapper = shallow(
      <HappyHourContainer locations={mockHappyHourInfo} />);
    
    expect(wrapper).toMatchSnapshot();
  });
});


