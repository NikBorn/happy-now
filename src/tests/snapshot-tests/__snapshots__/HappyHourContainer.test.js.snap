// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HappyHourContainer should match the snapshot everytime 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HappyHourContainer
    locations={
        Array [
            Object {
              "fireBaseId": "-KyD7aLGYd2eyMbtQgqa",
              "happyHours": Object {
                "Friday": Array [
                  "2:00PM",
                  "6:00PM",
                  "FREE BEER TOMORROW!",
                ],
                "Monday": Array [
                  "2:00PM",
                  "6:00PM",
                  "FREE BEER TOMORROW!",
                ],
                "Saturday": Array [
                  "2:00PM",
                  "6:00PM",
                  "FREE BEER TOMORROW!",
                ],
                "Sunday": Array [
                  "2:00PM",
                  "6:00PM",
                  "FREE BEER TOMORROW!",
                ],
                "Thursday": Array [
                  "2:00PM",
                  "6:00PM",
                  "FREE BEER TOMORROW!",
                ],
                "Tuesday": Array [
                  "2:00PM",
                  "6:00PM",
                  "FREE BEER TOMORROW!",
                ],
                "Wednesday": Array [
                  "2:00PM",
                  "6:00PM",
                  "FREE BEER TOMORROW!",
                ],
              },
              "locationId": "4b044f71f964a5202e5322e3",
              "name": "Paddy's Pub",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HappyHourCard
          happyHourInfo={
                    Object {
                              "fireBaseId": "-KyD7aLGYd2eyMbtQgqa",
                              "happyHours": Object {
                                "Friday": Array [
                                  "2:00PM",
                                  "6:00PM",
                                  "FREE BEER TOMORROW!",
                                ],
                                "Monday": Array [
                                  "2:00PM",
                                  "6:00PM",
                                  "FREE BEER TOMORROW!",
                                ],
                                "Saturday": Array [
                                  "2:00PM",
                                  "6:00PM",
                                  "FREE BEER TOMORROW!",
                                ],
                                "Sunday": Array [
                                  "2:00PM",
                                  "6:00PM",
                                  "FREE BEER TOMORROW!",
                                ],
                                "Thursday": Array [
                                  "2:00PM",
                                  "6:00PM",
                                  "FREE BEER TOMORROW!",
                                ],
                                "Tuesday": Array [
                                  "2:00PM",
                                  "6:00PM",
                                  "FREE BEER TOMORROW!",
                                ],
                                "Wednesday": Array [
                                  "2:00PM",
                                  "6:00PM",
                                  "FREE BEER TOMORROW!",
                                ],
                              },
                              "locationId": "4b044f71f964a5202e5322e3",
                              "name": "Paddy's Pub",
                            }
          }
/>,
      ],
      "className": "happy-hour-cntr",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "4b044f71f964a5202e5322e3",
        "nodeType": "function",
        "props": Object {
          "happyHourInfo": Object {
            "fireBaseId": "-KyD7aLGYd2eyMbtQgqa",
            "happyHours": Object {
              "Friday": Array [
                "2:00PM",
                "6:00PM",
                "FREE BEER TOMORROW!",
              ],
              "Monday": Array [
                "2:00PM",
                "6:00PM",
                "FREE BEER TOMORROW!",
              ],
              "Saturday": Array [
                "2:00PM",
                "6:00PM",
                "FREE BEER TOMORROW!",
              ],
              "Sunday": Array [
                "2:00PM",
                "6:00PM",
                "FREE BEER TOMORROW!",
              ],
              "Thursday": Array [
                "2:00PM",
                "6:00PM",
                "FREE BEER TOMORROW!",
              ],
              "Tuesday": Array [
                "2:00PM",
                "6:00PM",
                "FREE BEER TOMORROW!",
              ],
              "Wednesday": Array [
                "2:00PM",
                "6:00PM",
                "FREE BEER TOMORROW!",
              ],
            },
            "locationId": "4b044f71f964a5202e5322e3",
            "name": "Paddy's Pub",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HappyHourCard
            happyHourInfo={
                        Object {
                                    "fireBaseId": "-KyD7aLGYd2eyMbtQgqa",
                                    "happyHours": Object {
                                      "Friday": Array [
                                        "2:00PM",
                                        "6:00PM",
                                        "FREE BEER TOMORROW!",
                                      ],
                                      "Monday": Array [
                                        "2:00PM",
                                        "6:00PM",
                                        "FREE BEER TOMORROW!",
                                      ],
                                      "Saturday": Array [
                                        "2:00PM",
                                        "6:00PM",
                                        "FREE BEER TOMORROW!",
                                      ],
                                      "Sunday": Array [
                                        "2:00PM",
                                        "6:00PM",
                                        "FREE BEER TOMORROW!",
                                      ],
                                      "Thursday": Array [
                                        "2:00PM",
                                        "6:00PM",
                                        "FREE BEER TOMORROW!",
                                      ],
                                      "Tuesday": Array [
                                        "2:00PM",
                                        "6:00PM",
                                        "FREE BEER TOMORROW!",
                                      ],
                                      "Wednesday": Array [
                                        "2:00PM",
                                        "6:00PM",
                                        "FREE BEER TOMORROW!",
                                      ],
                                    },
                                    "locationId": "4b044f71f964a5202e5322e3",
                                    "name": "Paddy's Pub",
                                  }
            }
/>,
        ],
        "className": "happy-hour-cntr",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "4b044f71f964a5202e5322e3",
          "nodeType": "function",
          "props": Object {
            "happyHourInfo": Object {
              "fireBaseId": "-KyD7aLGYd2eyMbtQgqa",
              "happyHours": Object {
                "Friday": Array [
                  "2:00PM",
                  "6:00PM",
                  "FREE BEER TOMORROW!",
                ],
                "Monday": Array [
                  "2:00PM",
                  "6:00PM",
                  "FREE BEER TOMORROW!",
                ],
                "Saturday": Array [
                  "2:00PM",
                  "6:00PM",
                  "FREE BEER TOMORROW!",
                ],
                "Sunday": Array [
                  "2:00PM",
                  "6:00PM",
                  "FREE BEER TOMORROW!",
                ],
                "Thursday": Array [
                  "2:00PM",
                  "6:00PM",
                  "FREE BEER TOMORROW!",
                ],
                "Tuesday": Array [
                  "2:00PM",
                  "6:00PM",
                  "FREE BEER TOMORROW!",
                ],
                "Wednesday": Array [
                  "2:00PM",
                  "6:00PM",
                  "FREE BEER TOMORROW!",
                ],
              },
              "locationId": "4b044f71f964a5202e5322e3",
              "name": "Paddy's Pub",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
